CODING_RULES.md

```diff
@@ -216,3 +216,96 @@ Not part of the caption anymore etc.
                 for computing Precision, recall, ...
         """
 ```
+
+
+## Code comments
+
+### FIXME, TODO
+
+Feature branch code comments sometimes contain:
+
+- `FIXME`: potential bug, depending on other feature (so it cannot efficiently be solved now)
+- `TODO`: something to do in general
+
+`FIXME` and `TODO` in comments:
+
+- should be discussed during review for merging to `develop`
+- if kept, an associated issue must be created and referenced (eg `issue #1234`) in the comment
+
+
+## Branches
+
+Goals for regulating branches usage:
+
+- reach simple and informative history (limit entry number, remove hard to follow graphs)
+- keep it simple and safe for developers
+
+These guidelines address the commit/push/pull/merge aspects of branches. Lifecycle guidelines are [available here](./docs/developer/usage_and_tools.md#lifecycle).
+
+### comments in commit
+
+As a general rule, for comments in commit:
+
+- first line should give big picture of the commit
+- optional second line should be blank (to avoid git being confused)
+- optional later lines give commit details
+
+Do:
+```
+Add new function xxx
+
+- implement item1
+- improve item2
+- remove item3
+- fix item4
+- item5 not yet working
+```
+
+### commit, push, pull
+
+These guidelines mostly apply when working in a feature branch.
+
+Commit, push:
+- work freely in your local branch (including using micro-commits, etc.).
+- shrink contribution's history to a few significant commits *before pushing* (avoid pushing micro-commits or successive versions of same code to the remote)
+  - can use `git commit --amend`
+  - can squash local commits with `git rebase -i local_branch_base_commit` to shrink the yet-unpushed commits from `local_branch_base_commit`. Typically keep the first commit as `pick`, don't change order of commits, turn the next ones to `squash` to merge them in a single commit.
+  - stick to squashing the latest commits (unless you really know what you are doing)
+
+Pull:
+- can always use `git pull --rebase` (or add it to configuration to apply by default). No danger, it only rebases local yet-unpushed commits on top of new pulled remote commits.
+
+### update, merge
+
+These guidelines mostly apply when updating a feature branch or merging it to `develop`.
+
+Update a feature branch with latest version of `develop`:
+- nice to rebase feature branch from develop (`git rebase develop`) as explained below, rather than merging develop in feature branch (`git merge develop`)
+- caveat: don't mix rebase and merging for successive updates of a feature branch !
+
+Merge:
+- before merging, if too many commits were pushed to the feature branch, squash *remote branch* with a rebase (see below)
+
+### rebase remote branch
+
+These guidelines mostly apply when updating a feature branch or merging it to `develop`.
+- don't rebase on `develop` and `master` branches. Merge to these branches using a merge commit (default in github).
```

Comment:
Thank you @sharkovsky for your review !

I'm not really sure about this wording change ( "don't rebase `develop` and `master` branches on your feature branch"). I'm not sure this action would make sense ...
Weren't we just wanting to mention "don't rebase `develop` and `master` branches" (with no *on*) because we don't want any re-write of these branches' history ?

We can re-discuss it to reach a conclusion about this one ...